src/commands/remove.rs

```diff
@@ -0,0 +1,134 @@
+use crate::{
+    colors::ColorChoiceExt,
+    commands::GlobalArgs,
+    dirs::{init_venv, pyproject_path, read_pyproject},
+    error::{self, Result},
+    python::pip_install,
+    revert_file::RevertFile,
+};
+use aqora_config::Requirement;
+use aqora_runner::python::{PipOptions, PipPackage};
+use clap::Args;
+use indicatif::ProgressBar;
+use std::time::Duration;
+use tokio::fs;
+use toml_edit::DocumentMut;
+
+#[derive(Args, Debug)]
+#[command(author, version, about)]
+pub struct Remove {
+    pub deps: Vec<String>,
+}
+
+pub async fn remove(args: Remove, global: GlobalArgs) -> Result<()> {
+    let mut deps = Vec::new();
+    for dep in args.deps.iter() {
+        let req = Requirement::parse(dep, &global.project)
+            .map_err(|e| error::user(&format!("Invalid requirement '{dep}': {e}"), ""))?;
+        if !req.extras.is_empty() || req.version_or_url.is_some() || req.marker.is_some() {
+            return Err(error::user(
+                &format!("Invalid requirement '{dep}'"),
+                "Only the package name is allowed",
+            ));
+        }
+        deps.push(req);
+    }
+    let _ = read_pyproject(&global.project).await?;
+    let progress = ProgressBar::new_spinner();
+    progress.set_message("Initializing virtual environment");
+    progress.enable_steady_tick(Duration::from_millis(100));
+    let env = init_venv(&global.project, global.uv.as_ref(), &progress, global.color).await?;
+    let project_file = RevertFile::save(pyproject_path(&global.project))?;
+    let mut toml = fs::read_to_string(&project_file)
+        .await?
+        .parse::<DocumentMut>()?;
+    let project = if let Some(project) = toml.get_mut("project") {
+        project
+    } else {
+        progress.finish_with_message("No dependencies to remove");
+        return Ok(());
+    };
+    let dependencies = if let Some(dependencies) = project
+        .as_table_mut()
+        .ok_or_else(|| {
+            error::user(
+                "Invalid pyproject.toml",
+                "The 'project' section must be a table",
+            )
+        })?
+        .get_mut("dependencies")
+    {
+        dependencies
+    } else {
+        progress.finish_with_message("No dependencies to remove");
+        return Ok(());
+    };
+    let dependencies = dependencies.as_array_mut().ok_or_else(|| {
+        error::user(
+            "Invalid pyproject.toml",
+            "The 'dependencies' section must be an array",
+        )
+    })?;
+    let mut removed_deps = Vec::new();
+    for dep in deps.iter() {
+        let mut removed = false;
+        while let Some(index) = dependencies
```

Comment:
I think it would have better performances to traverse once the project dependencies, and traverse remaining parameters, like this:

```rust
let dependencies = dependencies.iter()
  .filter(|dependency| {
    if let Some(index) = deps.iter().position(|dep| dep.name == dependency.name) {
      deps.swap_remove(index); // this is O(1), and we don't need deps to keep ordering
      false
    }
    true
  })
  .collect::<Vec<_>>();
```

Reply:
i agree, that would be more performant, but the reason i preserve the order has more to do with the toml whitespace than anything else, but in any case i'm not too worried about performance because the list is probably never going to be more than the order of 100 items. but i did update the function to be somewhat more performant by only parsing once